src/libsync/creds/oauth.cpp

```diff
@@ -79,23 +78,16 @@ class RegisterClientJob : public QObject
 
     void start()
     {
-        auto keychainJob = new QKeychain::ReadPasswordJob(Theme::instance()->appName());
-        keychainJob->setKey(AbstractCredentials::keychainServerWideKey(_account, clientSecretC()));
-        connect(keychainJob, &QKeychain::ReadPasswordJob::finished, this, [this, keychainJob] {
-            if (keychainJob->error() == QKeychain::NoError) {
-                QJsonParseError error;
-                const auto doc = QJsonDocument::fromJson(keychainJob->binaryData(), &error);
-                if (error.error == QJsonParseError::NoError) {
-                    rgisterClientFinished(doc);
-                    return;
-                }
-                qCWarning(lcOauth) << Q_FUNC_INFO << error.errorString();
-            } else if (keychainJob->error() != QKeychain::EntryNotFound) {
-                qCCritical(lcOauth) << "Failed to read client id" << keychainJob->errorString();
+        auto job = _account->credentialManager()->getCredentials(clientSecretC());
```

Comment:
Why do export the credentialsManager() at all? And not use `_account->getCredentials(...)` ?

Reply:
Hm instead of exporting the credential manager provide a set and get method? We wouldn't really gain much?